src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformStringTest.cs

```diff
@@ -0,0 +1,389 @@
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.InteropServices.UseValidPlatformString,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.InteropServices.UseValidPlatformString,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices.UnitTests
+{
+    public class UseValidPlatformStringTest
+    {
+        private const string s_msBuildPlatforms = "build_property._SupportedPlatformList=CustomPlatform";
+
+        [Fact]
+        public async Task ValidPlatformStringUsedNotWarn()
+        {
+            var csSource = @"
+using System;
+using System.Runtime.Versioning;
+
+public class Test
+{
+    [SupportedOSPlatform(""Browser"")]
+    public void SupportedBrowser() { }
+
+    [SupportedOSPlatform(""iOS"")]
+    public string SupportedIos => string.Empty;
+
+    [UnsupportedOSPlatform(""linux"")]
+    public void UnsupportedLinux() { }
+
+    [SupportedOSPlatform(""Android"")]
+    public event EventHandler AndroidOnlyEvent;
+
+    [UnsupportedOSPlatform(""macos"")]
+    public void UnsupportedMacOS() { }
+
+    [SupportedOSPlatform(""TVOS"")]
+    public Test() { }
+
+    [SupportedOSPlatform(""freeBsd2.0.1.2"")]
+    public void SupportedFreeBsd() { }
+
+    [SupportedOSPlatform(""watchOS3.2.1"")]
+    public void SupporteWatchOS() { }
+
+    [UnsupportedOSPlatform(""windows7.0"")]
+    public static void SupportedWindows() { }
+}";
+            await VerifyAnalyzerAsyncCs(csSource);
+        }
+
+        [Fact]
+        public async Task ValidPlatformStringUsedNotWarnVB()
+        {
+            var vbSource = @"
+Imports System.Runtime.Versioning
+
+Public Class Test
+    <SupportedOSPlatform(""Browser"")>
+    Public Sub SupportedBrowser()
+    End Sub
+
+    <SupportedOSPlatform(""iOS"")>
+    Public Sub SupportedIos()
+    End Sub
+
+    <UnsupportedOSPlatform(""linux"")>
+    Public Sub UnsupportedLinux()
+    End Sub
+
+    <SupportedOSPlatform(""Android"")>
+    Public Sub SupportedAndroid()
+    End Sub
+
+    <UnsupportedOSPlatform(""macos"")>
+    Public Sub UnsupportedMacOS()
+    End Sub
+
+    <SupportedOSPlatform(""tvos"")>
+    Public Sub SupportedTvOS()
+    End Sub
+
+    <SupportedOSPlatform(""freeBsd2.0.1.2"")>
+    Public Sub SupportedFreeBsd()
+    End Sub
+
+    <SupportedOSPlatform(""watchOS3.2.1"")>
+    Public Sub SupporteWatchOS()
+    End Sub
+
+    <UnsupportedOSPlatform(""WINDOWS7.0"")>
+    Public Shared Sub SupportedWindows()
+    End Sub
+End Class";
+            await VerifyAnalyzerAsyncVb(vbSource);
+        }
+
+        [Fact]
+        public async Task InvalidPlatformNameOrVersionWarns()
+        {
+            var csSource = @"
+using System.Runtime.Versioning;
+
+public class Test
+{
+    [[|UnsupportedOSPlatform(""window7.0"")|]] // The platform 'window' is not a known platform name
```

Comment:
Did not found a way for reporting the diagnostics at the attribute's argument's location. Seems I could do some workaround like creating a new location by subtracting the attribute name length or something like that, but not sure if that is the desired approach, any suggestions @mavasani?

Reply:
I think showing the diagnostic on the entire attribute is just fine.